src/Components/Common/MarkdownEditor.tsx

```diff
@@ -0,0 +1,396 @@
+    if (!selection || !selection.rangeCount) return;
+
+    const range = selection.getRangeAt(0);
+    const blockquote = document.createElement("blockquote");
+    blockquote.appendChild(range.extractContents());
+    range.insertNode(blockquote);
+
+    const br = document.createElement("br");
+    if (blockquote.parentNode) {
+      blockquote.parentNode.insertBefore(br, blockquote.nextSibling);
+    }
+
+    saveState();
+  };
+
+  const handleLink = () => {
+    const userLink = prompt("Enter a URL");
+    if (userLink) {
+      const selection = window.getSelection();
+      if (!selection || !selection.rangeCount) return;
+
+      const range = selection.getRangeAt(0);
+      const anchor = document.createElement("a");
+      anchor.href = userLink.startsWith("http")
+        ? userLink
+        : `http://${userLink}`;
+      anchor.textContent = range.toString();
+
+      range.deleteContents();
+      range.insertNode(anchor);
+      saveState();
+    }
+  };
+
+  const handleUnlink = () => {
+    const selection = window.getSelection();
+    if (!selection || !selection.rangeCount) return;
+
+    const range = selection.getRangeAt(0);
+    const container = range.startContainer.parentNode as HTMLElement;
+    if (container && container.tagName === "A") {
+      const parent = container.parentNode;
+      while (container.firstChild) {
+        if (parent) {
+          parent.insertBefore(container.firstChild, container);
+        }
+      }
+      if (parent) {
+        parent.removeChild(container);
+      }
+    }
+    saveState();
+  };
+
+  const handleLineBreak = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === "Enter") {
+      e.preventDefault();
+      const selection = window.getSelection();
+      if (!selection) return;
+
+      const range = selection.getRangeAt(0);
+      const container = range.startContainer.parentNode as HTMLElement;
+
+      if (container && container.tagName === "LI") {
+        const listNode = container.parentNode as HTMLElement;
+        if (listNode) {
+          const newList = document.createElement(listNode.tagName);
+          const newListItem = document.createElement("li");
+
+          newListItem.appendChild(range.extractContents());
+          newList.appendChild(newListItem);
+          if (listNode.parentNode) {
+            listNode.insertBefore(newListItem, container.nextSibling);
+          }
+
+          range.setStart(newListItem, 0);
+          range.setEnd(newListItem, 0);
+
+          if (
+            container &&
+            container.textContent &&
+            container.textContent.trim() === ""
+          ) {
+            if (listNode.parentNode) {
+              listNode.removeChild(container);
+            }
+          }
+        }
+      } else {
+        if (isUnorderedListActive || isOrderedListActive) {
+          document.execCommand("insertHTML", false, "<br><br>");
+        } else {
+          document.execCommand("insertLineBreak");
```

Comment:
`document.execCommand( ... )` is deprecated, consider other alternatives

Reply:
working on it !